src/main/java/net/imglib2/loops/BindActionToSamplers.java

```diff
@@ -272,4 +304,509 @@ public void run()
+
+		private final RandomAccess< H > accessH;
+
+		private final RandomAccess< I > accessI;
+
+		public NineConsumerLocalizingRunnable( final LoopBuilder.NineConsumer< RandomAccess< A >, RandomAccess< B >, RandomAccess < C >, RandomAccess< D >, RandomAccess< E >, RandomAccess< F >, RandomAccess< G >, RandomAccess< H >, RandomAccess< I > > action, final RandomAccess< A > accessA, final RandomAccess< B > accessB, final RandomAccess< C > accessC, final RandomAccess< D > accessD, final RandomAccess< E > accessE, final RandomAccess< F > accessF, final RandomAccess< G > accessG, final RandomAccess< H > accessH, final RandomAccess< I > accessI )
+		{
+			this.action = action;
+			this.accessA = accessA;
+			this.accessB = accessB;
+			this.accessC = accessC;
+			this.accessD = accessD;
+			this.accessE = accessE;
+			this.accessF = accessF;
+			this.accessG = accessG;
+			this.accessH = accessH;
+			this.accessI = accessI;
+		}
+
+		@Override
+		public void run()
+		{
+			action.accept( accessA, accessB, accessC, accessD, accessE, accessF, accessG, accessH, accessI );
+		}
+	}
+
+	public static class TenConsumerLocalizingRunnable< A, B, C, D, E, F, G, H, I, J > implements Runnable
+	{
+
+		private final LoopBuilder.TenConsumer< RandomAccess< A >, RandomAccess< B >, RandomAccess< C >, RandomAccess< D >, RandomAccess< E >, RandomAccess< F >, RandomAccess< G >, RandomAccess< H >, RandomAccess< I >, RandomAccess< J > > action;
+
+		private final RandomAccess< A > accessA;
+
+		private final RandomAccess< B > accessB;
+
+		private final RandomAccess< C > accessC;
+
+		private final RandomAccess< D > accessD;
+
+		private final RandomAccess< E > accessE;
+
+		private final RandomAccess< F > accessF;
+
+		private final RandomAccess< G > accessG;
+
+		private final RandomAccess< H > accessH;
+
+		private final RandomAccess< I > accessI;
+
+		private final RandomAccess< J > accessJ;
+
+		public TenConsumerLocalizingRunnable( final LoopBuilder.TenConsumer< RandomAccess< A >, RandomAccess< B >, RandomAccess < C >, RandomAccess< D >, RandomAccess< E >, RandomAccess< F >, RandomAccess< G >, RandomAccess< H >, RandomAccess< I >, RandomAccess< J > > action, final RandomAccess< A > accessA, final RandomAccess< B > accessB, final RandomAccess< C > accessC, final RandomAccess< D > accessD, final RandomAccess< E > accessE, final RandomAccess< F > accessF, final RandomAccess< G > accessG, final RandomAccess< H > accessH, final RandomAccess< I > accessI, final RandomAccess< J > accessJ )
+		{
+			this.action = action;
+			this.accessA = accessA;
+			this.accessB = accessB;
+			this.accessC = accessC;
+			this.accessD = accessD;
+			this.accessE = accessE;
+			this.accessF = accessF;
+			this.accessG = accessG;
+			this.accessH = accessH;
+			this.accessI = accessI;
+			this.accessJ = accessJ;
+		}
+
+		@Override
+		public void run()
+		{
+			action.accept( accessA, accessB, accessC, accessD, accessE, accessF, accessG, accessH, accessI, accessJ );
+		}
+	}
```

Comment:
The implementation of LoopBuilder has already many lines of code. I fear that adding these 500+ lines of code reduces the maintainability. Let us go for this shorter solution https://github.com/imglib/imglib2/pull/279/.